src/controllers/wishlistController.js

```diff
@@ -4,19 +4,47 @@ const { StatusCodes } = require('http-status-codes');
 
 //Create wishlist
 const createWishlist = async (req, res) => {
-  const existingWishlist = await Wishlist.findOne({ user: req.user.userId });
+  const existingWishlist = await Wishlist.findOne({ user: req.user.userId })
+  .populate('albums') // Populate the 'albums' field
+      .exec();
 
   if (!existingWishlist) {
     const newWishlist = new Wishlist({
       user: req.user.userId,
-      albums: [], // start with an empty wishlist
+      albums: req.body.albums || [], // start with an empty wishlist
     });
 
     await newWishlist.save();
```

Comment:
The returned `newWishlist` won't have full albums populated just the ids.

We can do a query here to fetch/populate the full album bodies after saving them:

```suggestion
    await newWishlist.save();
    await newWishlist.populate('albums');
```